### PR TITLE
Enable more pbft tests

### LINKED ISSUE #934: Check Raft only tests
After #562 we noticed that there are some e2e tests that run only with Raft. It would be good to go over them and see which ones can and should be run also with PBFT

### PR DESCRIPTION
**Pull request by olgavrou**
Resolves #934 

Tests that are still not enabled are the recovery, reconfiguration related tests, code update test.

- `lua_end_to_end_batched` messages exceed `Max_message_size` so still can't be enabled

Tests that can't be enabled because they raise issues that need to be investigated and addressed separately (will open tickets) are:
 
- `lua_txregulator_test` and `js_end_to_end_logging` tests trigger a pbft view change which causes the replicas to roll back before the last checkpoint which in turn crashes the nodes #951 
- `lua_end_to_end_logging` and `end_to_end_scenario` are unstable (tagging them along with the above issue)
- `member_clients` test makes merkle roots between replicas un-matched #952
- `governance_history` test fails because it checks the ledger for governance results, and in pbft the governance maps are derived and do not go in the ledger #953

### CODE REVIEW

File: samples/apps/smallbank/smallbank.cmake [hunk 1]
```diff
@@ -49,38 +61,26 @@ if(BUILD_TESTS)
...
-
-  if(${SERVICE_IDENTITY_CURVE_CHOICE} STREQUAL "secp256k1_bitcoin")
-    set(SMALL_BANK_SIGNED_VERIFICATION_FILE
-        ${CMAKE_CURRENT_LIST_DIR}/tests/verify_small_bank_50k.json
-    )
-    set(SMALL_BANK_SIGNED_ITERATIONS 50000)
-  else()
-    set(SMALL_BANK_SIGNED_VERIFICATION_FILE
-        ${CMAKE_CURRENT_LIST_DIR}/tests/verify_small_bank_2k.json
+    # These tests require client-signed signatures: - HTTP C++ perf clients
```
Comment (jumaffre) [inline samples/apps/smallbank/smallbank.cmake]: I know we've been discussing this before but I believe that this now works with PBFT?

Comment (jumaffre) [reply samples/apps/smallbank/smallbank.cmake]: I think the request signatures are only verified on the forwardee node though.

**Comment by cimetrics** [comment]
enable_more_pbft_tests@[6025 aka 20200313.22](https://msrc-ccf.visualstudio.com/CCF/_build/results?buildId=6025&view=results) vs master ewma over 30 builds from [5749](https://msrc-ccf.visualstudio.com/CCF/_build/results?buildId=5749&view=results) to [6018](https://msrc-ccf.visualstudio.com/CCF/_build/results?buildId=6018&view=results)
images

File: samples/apps/smallbank/smallbank.cmake (hunk 1 above)
Comment (olgavrou) [reply samples/apps/smallbank/smallbank.cmake]: PBFT skips forwarding though so I don't think there is a point in running this test

Comment (jumaffre) [reply samples/apps/smallbank/smallbank.cmake]: PBFT uses its own forwarding path. However, this is the only performance we have with client signatures so I think we should enable it?

Comment (jumaffre) [reply samples/apps/smallbank/smallbank.cmake]: Sorry I got confused between this test and `small_bank_sigs_forwarding`. This comment should go though.

File: tests/connections.py [hunk 2]
```diff
@@ -23,7 +23,7 @@
 
 
 def run(args):
-    hosts = ["localhost"]
+    hosts = ["localhost"] * (4 if args.consensus == "pbft" else 1)
```
Comment (jumaffre) [inline tests/connections.py]: Shouldn't this work with one node in PBFT (`f = 0`)?

File: tests/e2e_logging.py [hunk 3]
```diff
@@ -15,12 +15,18 @@
 @reqs.at_least_n_nodes(2)
 def test(network, args, notifications_queue=None, verify=True):
     txs = app.LoggingTxs(notifications_queue=notifications_queue)
-    txs.issue(network=network, number_txs=1, wait_for_sync=args.consensus == "raft")
+    txs.issue(
+        network=network,
+        number_txs=1,
+        wait_for_sync=args.consensus == "raft",
+        consensus=args.consensus,
```
Comment (jumaffre) [inline tests/e2e_logging.py]: There's already a `wait_for_sync` argument that depends on the consensus. Can you collapse the two arguments?

File: tests/governance.py [hunk 4]
```diff
@@ -20,13 +20,13 @@
 
 
 def run(args):
-    hosts = ["localhost", "localhost"]
+    hosts = ["localhost"] * (4 if args.consensus == "pbft" else 2)
```
Comment (jumaffre) [inline tests/governance.py]: Could the number of hosts be automatically derived from `args.consensus` without having to specify it for each test?

File: tests/infra/ccf.py [hunk 5]
```diff
@@ -486,7 +486,13 @@ def wait_for_node_commit_sync(self, timeout=3):
             if [commits[0]] * len(commits) == commits:
                 break
             time.sleep(1)
-        assert [commits[0]] * len(commits) == commits, "All nodes at the same commit"
+        # in pbft getCommit acts as a write, so commits will not be the same
```
Comment (jumaffre) [inline tests/infra/ccf.py]: "acts as a write" -> increment the commit index?

File: samples/apps/smallbank/smallbank.cmake (hunk 1 above)
Comment (achamayou) [reply samples/apps/smallbank/smallbank.cmake]: We should still run a test with client signatures against PBFT, it's useful to know what the performance of that is, whether the signatures can be offloaded or not.

Comment (jumaffre) [reply samples/apps/smallbank/smallbank.cmake]: Yep. This test is enabled with PBFT now though. 
image

Comment (achamayou) [reply samples/apps/smallbank/smallbank.cmake]: Oh right, sorry about that then.

File: tests/connections.py (hunk 2 above)
Comment (olgavrou) [reply tests/connections.py]: if the network is openend f becomes 1 though

File: tests/e2e_logging.py (hunk 3 above)
Comment (olgavrou) [reply tests/e2e_logging.py]: We could collapse the two arguments, but will there be a case where we want to set wait for sync independently from the consensus being used?

Comment (jumaffre) [reply tests/e2e_logging.py]: The `wait_for_sync` flag was added as a quick fix for PBFT so we can remove it safely.

Comment (olgavrou) [reply tests/e2e_logging.py]: awesome

**Pull request merged**